File: nrepl-plugin/src/main/java/ucdenver/ccp/protege/nrepl/OWLOntologyChangeListenerImpl.java
package ucdenver.ccp.protege.nrepl;

import org.semanticweb.owlapi.model.OWLException;
import org.semanticweb.owlapi.model.OWLOntologyChange;
import org.semanticweb.owlapi.model.OWLOntologyChangeListener;

import javax.annotation.Nonnull;
import java.util.List;

public class OWLOntologyChangeListenerImpl implements OWLOntologyChangeListener {
    @Override
    public void ontologiesChanged(@Nonnull List<? extends OWLOntologyChange> list) throws OWLException {

    }
}

File: nrepl-clojure/src/protege/OWLOntologyChangeListenerImpl.clj
(ns protege.OWLOntologyChangeListenerImpl
  (:gen-class
   :name protege.OWLOntologyChangeListenerImpl
   :main false
   :implements [OWLOntologyChangeListener]
   :methods [[ontologiesChanged [List] void]]))

(defn -ontologiesChanged [list]
  (println (bean list)))
